.github/workflows/check-and-gen.yml

```diff
@@ -70,3 +68,37 @@ jobs:
       - uses: ./.github/actions/install-go
       - run: go mod tidy -v
       - run: git diff --exit-code
+  generate-docs:
+    name: Generate Documentation
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/'))
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+
+      - uses: ./.github/actions/install-system-dependencies
+      - uses: ./.github/actions/install-go
+      - uses: ./.github/actions/make-deps
+
+      - name: Generate API documentation using docsgen-cli
+        run: |
+          make docsgen-cli || {
+            echo "Error: Documentation generation failed"
+            exit 1
+          }
+
+      - name: Commit and push if documentation changed
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            git config user.name "github-actions[bot]"
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git add .
+            git commit -m "docs: update API documentation via docsgen-cli"
```

Comment:
If we're adding a commit, I think it would be good to tie what PR this commit is asociated with. 
(We haven't developed team standards yet on whether commits should have a PR reference, but I think this is coming and we may as well do it here.)
Maybe @rvagg has input?
(This isn't blocking though.)

Reply:
This can be disregarded.  I understand that this commit will be added before the PR is merged, and since  most PRs will be squashed, this goes away.  I'll resolve this.